homeassistant/components/fyta/__init__.py

```diff
@@ -47,3 +60,36 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version > 1:
+        # This means the user has downgraded from a future version
+        return False
+
+    if config_entry.version == 1:
+        new = {**config_entry.data}
+        if config_entry.minor_version < 2:
+            fyta = FytaConnector(
+                config_entry.data[CONF_USERNAME], config_entry.data[CONF_PASSWORD]
+            )
+            credentials: dict[str, Any] = await fyta.login()
+            await fyta.client.close()
+
+            new[CONF_ACCESS_TOKEN] = credentials[CONF_ACCESS_TOKEN]
+            new[CONF_EXPIRATION] = credentials[CONF_EXPIRATION].isoformat()
+
+        hass.config_entries.async_update_entry(
```

Comment:
We only need to migrate the config entry if the minor version is lower than the wanted minor version.

Reply:
Correct.
I assumed, `async_migrate_entry` is only called, if we have a newer config version available than in the config entry. Is this not correct and it's called on every startup?
Anyway, I can update this.